jitsi-utils-kotlin/src/main/kotlin/org/jitsi/utils/properties/Interfaces.kt

```diff
@@ -18,20 +18,94 @@ package org.jitsi.utils.properties
 
 import kotlin.reflect.KProperty
 
+/**
+ * Base interface that can be used for implementing property delegates of read-only properties
+ * of int primitive type.
+ *
+ * This is provided only for convenience; you don't have to extend this interface
+ * as long as your property delegate has methods with the same signatures.
+ *
+ * @param T the type of object which owns the delegated property.
+ */
 public interface ReadOnlyIntProperty<in T> {
```

Comment:
Shouldn't these be non-generic?

Reply:
@bbaldino not sure, original generic type has 2 generic parameters, 1 to describe owner type and 1 to describe type of value stored in property. So I've basically [copy-pasted original code from `kotlin.properties`](https://github.com/JetBrains/kotlin/blob/master/libraries/stdlib/src/kotlin/properties/Interfaces.kt) and replaced type argument of property type with change in name of the type.

Reply:
Ok, I guess we could prove this out by comparing the allocations rates before/after (and also seeing if this spot is causing an allocation).  Did you get a chance to try that?

Reply:
@bbaldino yes, sure. I'll attach screen recording of Visual VM with observation I had before and after changes for a video conference of 10 peers with 640x480 video, simulcast enabled and LastN=1. Does it sound good? I'm was almost done with videos, but was interrupted to [proposing changes to Kotlin library](https://github.com/JetBrains/kotlin/pull/3320).

Reply:
@bbaldino I'm really sorry, I won't be able to provide measurements today. I'm just run out of time today :( and also probably thought of an alternative and probably better idea how to fix an issue.

Reply:
No problem